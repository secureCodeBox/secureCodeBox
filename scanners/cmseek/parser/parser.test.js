// SPDX-FileCopyrightText: the secureCodeBox authors
//
// SPDX-License-Identifier: Apache-2.0

const fs = require("fs");
const util = require("util");

const readFile = util.promisify(fs.readFile);

const { parse } = require("./parser");

const {
  validateParser,
} = require("@securecodebox/parser-sdk-nodejs/parser-utils");

test("parser parses result of Joomla scan with core vulnerabilities successfully", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/joomla_with_core_vulns.json",
    {
      encoding: "utf8",
    }
  );

  const findings = await parse(JSON.parse(fileContent));
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parser parses result of Joomla scan without core vulnerabilities successfully", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/joomla_without_core_vulns.json",
    {
      encoding: "utf8",
    }
  );

  const findings = await parse(JSON.parse(fileContent));
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parser parses result of non-Joomla scan successfully", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/not_joomla.json",
    {
      encoding: "utf8",
    }
  );

  const findings = await parse(JSON.parse(fileContent));
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("should properly parse empty cmseek json file", async () => {
  const jsonContent = await readFile(
    __dirname + "/__testFiles__/test-empty-report.json",
    {
      encoding: "utf8",
    }
  );
  const findings = await parse(jsonContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchInlineSnapshot(`[]`);
});
