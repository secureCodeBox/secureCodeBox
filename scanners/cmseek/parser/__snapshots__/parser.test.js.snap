// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser parses result of Joomla scan with core vulnerabilities successfully 1`] = `
[
  {
    "attributes": {
      "joomla_version": "3.6.5",
      "references": [
        "CVE : CVE-2016-10033",
        "https://www.rapid7.com/db/modules/exploit/multi/http/phpmailer_arg_injection",
        "https://github.com/opsxcq/exploit-CVE-2016-10033",
        "EDB : https://www.exploit-db.com/exploits/40969/",
      ],
    },
    "category": "Vulnerability",
    "description": "Vulnerability of type PHPMailer Remote Code Execution Vulnerability found",
    "location": "http://172.26.0.3/",
    "name": "PHPMailer Remote Code Execution Vulnerability",
    "osi_layer": "APPLICATION",
    "severity": "HIGH",
  },
  {
    "attributes": {
      "joomla_version": "3.6.5",
      "references": [
        "CVE : CVE-2016-10045",
        "https://www.rapid7.com/db/modules/exploit/multi/http/phpmailer_arg_injection",
        "EDB : https://www.exploit-db.com/exploits/40969/",
      ],
    },
    "category": "Vulnerability",
    "description": "Vulnerability of type PPHPMailer Incomplete Fix Remote Code Execution Vulnerability found",
    "location": "http://172.26.0.3/",
    "name": "PPHPMailer Incomplete Fix Remote Code Execution Vulnerability",
    "osi_layer": "APPLICATION",
    "severity": "HIGH",
  },
  {
    "attributes": {
      "joomla_backup_files": [
        "http://172.26.0.3/1.save",
        "http://172.26.0.3/1.tar.gz",
        "http://172.26.0.3/1.rar",
        "http://172.26.0.3/1.tar",
        "http://172.26.0.3/1.zip",
        "http://172.26.0.3/1.txt",
        "http://172.26.0.3/1.tgz",
        "http://172.26.0.3/1.tar.bz2",
        "http://172.26.0.3/1.gz",
        "http://172.26.0.3/1.tmp",
      ],
    },
    "category": "Visible internal files",
    "description": "Visible Backup files found",
    "location": "http://172.26.0.3/",
    "name": "Backup files",
    "osi_layer": "APPLICATION",
    "severity": "INFORMATIONAL",
  },
  {
    "category": "Security Misconfiguration",
    "description": "Debug mode is enabled on the site",
    "location": "http://172.26.0.3/",
    "name": "Debug mode",
    "osi_layer": "APPLICATION",
    "severity": "MEDIUM",
  },
]
`;

exports[`parser parses result of Joomla scan without core vulnerabilities successfully 1`] = `[]`;

exports[`parser parses result of non-Joomla scan successfully 1`] = `[]`;
